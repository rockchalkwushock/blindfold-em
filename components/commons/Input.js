import Fragment from './Fragment'

const Input = ({ name, fn, value }) => (
  <Fragment>
    <input name={name} onChange={e => fn(e)} type="text" value={value} />
  </Fragment>
)

export default Input
