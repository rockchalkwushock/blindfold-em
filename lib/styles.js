export default {
  colors: {
    main: '#734D84',
    text: '#B0BF6C',
    highLight: '#A974C2',
    lowLight: '#3A2A41'
  },
  fonts: {
    kalam: '"Kalam", cursive',
    permMarker: '"Permanent Marker", cursive'
  }
}
