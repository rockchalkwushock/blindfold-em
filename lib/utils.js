import moment from 'moment'

/**
 * @function format
 * @description Function for formating moment.duration()
 *
 * @see https://github.com/moment/moment/issues/1048#issue-19008383
 *
 * @param {Object} time
 * @returns {String} "0:12:00"
 */
export const format = time =>
  Math.floor(time.asHours()) +
  moment.utc(time.asMilliseconds()).format(':mm:ss')

export const timerState = {
  STOPPED: 0,
  START: 1
}
