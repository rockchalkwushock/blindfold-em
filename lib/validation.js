const activityRegex = /^[0-9A-Z.!,'\s]+$/i
const timeRegex = /^[0-9]+$/i
const userGuide = "lower & upper case characters, numbers, and (.!,')"

export default ({ form }) => {
  const errors = {}
  // use currentFrame of Wizard to check case.
  switch (form.currentFrame) {
    case 1:
      if (!form.activity) {
        errors.activity = 'You must submit an activity.'
      } else if (!activityRegex.test(form.activity)) {
        errors.activity = `${
          form.activity
        } is not a valid input. Please stick too: ${userGuide}.`
      }
      return errors
    case 2:
      if (!form.timer) {
        errors.timer = 'You must submit a timer for the activity.'
      } else if (!timeRegex.test(form.timer)) {
        errors.timer = `${
          form.timer
        } is not a valid input. Please only submit number values.`
      }
      return errors
    case 3:
      if (!form.cooldown) {
        errors.cooldown = 'You must submit a cooldown for the activity.'
      } else if (!timeRegex.test(form.cooldown)) {
        errors.cooldown = `${
          form.cooldown
        } is not a valid input. Please only submit number values.`
      }
      return errors
    default:
      return errors
  }
}
