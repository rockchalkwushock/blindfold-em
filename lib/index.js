import moment from 'moment'

import styles from './styles'

/**
 * @function format
 * @description Function for formating moment.duration()
 *
 * @see https://github.com/moment/moment/issues/1048#issue-19008383
 *
 * @param {Object} time
 * @returns {String} "0:12:00"
 */
export const format = time =>
  Math.floor(time.asHours()) +
  moment.utc(time.asMilliseconds()).format(':mm:ss')

export { styles }
