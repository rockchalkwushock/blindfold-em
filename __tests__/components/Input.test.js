import React from 'react'
import { mount } from 'enzyme'

import Input from '../../components/commons/Input'
import { mockTestFn } from '../../lib'

describe('Component: <Input />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(<Input name="activity" fn={mockTestFn} value="" />)
  })

  test('1. Should show initial prop values', () => {
    expect(wrapper.props().name).toEqual('activity')
    expect(wrapper.props().value).toEqual('')
  })

  test('2. Should show updated prop values', () => {
    wrapper.setProps({
      value: 'coding'
    })
    expect(wrapper.props().name).toEqual('activity')
    expect(wrapper.props().value).toEqual('coding')
  })
})
