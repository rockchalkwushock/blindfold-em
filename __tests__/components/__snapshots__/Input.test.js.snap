// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Input /> should mount without exploding 1`] = `
<Test>
  <Input
    fn={[Function]}
    name="activity"
    value=""
  >
    <Fragment>
      <input
        name="activity"
        onChange={[Function]}
        type="text"
        value=""
      />
    </Fragment>
  </Input>
</Test>
`;
